src/components/TimeSelect/TimeSelectPage.tsx

```diff
@@ -171,11 +190,89 @@ function TimeSelectPage() {
   };
 
   const onPopupCloseAndSubmit = () => {
-    // @ts-expect-error
     setGoogleCalIds(selectedPopupIds);
     setGcalPopupOpen(false);
   };
 
+  const onPopupCloseAutofillAndSubmit = () => {
+    if (selectedPopupIds === googleCalIds) {
+      // no need to make changes if selected calendars didn't change
+      setGcalPopupOpen(false);
+      return;
+    }
+
+    setShouldFillAvailability(true);
+    setIsFillingAvailability(true);
+    setGoogleCalIds(selectedPopupIds);
+  };
+
+  const fillAvailabilityNotInGCalEvents = (
+    parsedEvents: any[],
+    dates: calandarDate[]
+  ) => {
+    const userIndex = getCurrentUserIndex();
+    const oldCalendarState = {...calendarState};
+    const userAvailability = oldCalendarState[userIndex];
+
+    // @ts-expect-error
+    const startHour = calendarFramework.startTime.getHours();
```

Comment:
Let's start using types. I know I'm the worst offender with using //@ts-expect-error, but we should start moving toward using actual typing